frontend/lib/resolution/scope-queries.cpp

```diff
@@ -828,8 +822,17 @@ bool LookupHelper::doLookupInScope(const Scope* scope,
   bool goPastModules = (config & LOOKUP_GO_PAST_MODULES) != 0;
   bool onlyMethodsFields = (config & LOOKUP_ONLY_METHODS_FIELDS) != 0;
   bool checkExternBlocks = (config & LOOKUP_EXTERN_BLOCKS) != 0;
+  bool skipPrivateUseImport = (config & LOOKUP_SKIP_PRIVATE_USE_IMPORT) != 0;
   bool trace = (traceCurPath != nullptr && traceResult != nullptr);
 
+  IdAndVis::SymbolTypeFlags curFilter = 0;
+  if (skipPrivateVisibilities) {
+    curFilter |= IdAndVis::PUBLIC;
+  }
```

Comment:
It exists so we can revisit scopes for additional symbols when we looked for public symbols only but then later realize we can also search for private symbols.

See the PR message:

> ... This set is built from enum values in `IdAndVis`, and there is a bit dedicated to each as well as its opposite (so, we have `PUBLIC`, `PRIVATE`, `METHOD_OR_FIELD`, and `NOT_METHOD_OR_FIELD`). The reason to have `PRIVATE` and `NOT_METHOD_OR_FIELD` (instead of just relying on the absence of a bit) is that it allows us to express all filters through set bits, where 0 bits mean "don't care". Since `0` bits mean "don't care", we can optimize `OwnedIdsWithName::borrow`. First, `OwnedIdsWithName` tracks the bitwise `&` and bitwise `|` of all of the contained `IdAndVis`. Then, within `OwnedIdsWithName::borrow`, we can easily detect two common cases: 1) the filter matches all Ids and 2) the filter matches no Ids. This keeps the process of borrowing O(1) in more cases. Note that a filter with bits set for `PUBLIC` and `METHOD_OR_FIELD` is requesting only Ids that meet both constraints (ie. only ids that are both public and are methods/fields).